File: web/src/interfaces/PessoaDto.ts
import { empty_Guid } from "../utils/guid";

export class PessoaDto {
  id: string;
  nome: string = "";
  idade: number = 0;
  cpf: string = "";
  dataCriacaoModificacao: string = new Date().toISOString();

  constructor(
    id: string = empty_Guid,
    nome: string = "",
    idade: number = 0,
    cpf: string = "",
    dataCriacaoModificacao: string = new Date().toISOString(),
  ) {
    this.id = id;
    this.nome = nome;
    this.idade = idade;
    this.cpf = cpf;
    this.dataCriacaoModificacao = dataCriacaoModificacao;
  }
}

File: api/HomeBudget.API/HomeBudget.Application/DTOs/PessoaDtos/PessoaDto.cs
namespace HomeBudget.Application.DTOs.PessoaDtos
{
    public class PessoaDto
    {
        public Guid Id { get; set; }
        public string Nome { get; set; }
        public string Cpf { get; set; }
        public int Idade { get; set; }
        public DateTime DataCriacao { get; set; }
    }
}
